runtime/v2/shim/shim.go

```diff
@@ -291,17 +312,37 @@ func dumpStacks(logger *logrus.Entry) {
 
 type remoteEventsPublisher struct {
 	address string
+	conn    net.Conn
 	client  v1.EventsService
+	closed  chan struct{}
+}
+
+func (l *remoteEventsPublisher) Done() <-chan struct{} {
+	return l.closed
+}
+
+func (l *remoteEventsPublisher) Close() error {
+	err := l.conn.Close()
+	close(l.closed)
```

Comment:
Can this cause a panic? I see a deferred `Close` after a `Done` check. Seems `Close` should be protected from being called multiple times.

Reply:
fixed